nats/pub_command.go

```diff
@@ -85,33 +109,46 @@ func (c *pubCmd) publish(pc *kingpin.ParseContext) error {
 			return err
 		}
 
-		m, err := nc.RequestMsg(msg, timeout)
-		if err != nil {
-			return err
-		}
+		if c.respCount == 1 {
+			m, err := nc.RequestMsg(msg, timeout)
+			if err != nil {
+				return err
+			}
+			c.printResponse(m)
+		} else {
+			if msg.Reply == "" {
+				msg.Reply = nc.NewRespInbox()
+			}
 
-		if c.raw {
-			fmt.Println(string(m.Data))
+			msgChan := make(chan *nats.Msg, c.respCount)
```

Comment:
really wish this whole pattern was in nats.go instead, I have several implementations of it.

Reply:
Agreed.

Reply:
Created an issue in nats.go . will update this PR once the issue is closed